samples/apps/copilot-chat-app/webapi/CopilotChat/Skills/ChatSkills/SemanticChatMemoryExtractor.cs

```diff
@@ -34,20 +34,26 @@ internal static class SemanticChatMemoryExtractor
     /// <param name="options">The prompts options.</param>
     internal static async Task ExtractSemanticChatMemoryAsync(
         string chatId,
-        IKernel kernel,
         SKContext context,
-        PromptsOptions options)
+        PromptsOptions options,
+        IDictionary<string, ISKFunction> chatSkillPlugin,
+        IDictionary<string, int> chatSkillTokenCounts)
     {
-        foreach (var memoryName in options.MemoryMap.Keys)
+        var memoryExtractionContext = Utilities.CopyContextWithVariablesClone(context);
+        memoryExtractionContext.Variables.Set("format", options.MemoryFormat);
+
+        foreach (var memoryName in options.MemoryTypes)
```

Comment:
nit: for consistency, use `memoryType` (singular of collection you're referencing MemoryTypes)

same for below